### ISSUE TITLE
Improve error handling for request retries

### ISSUE THREAD

**Issue by B4nan** [apify/apify-ts#32]
Currently every failed request ends up with a huge and scary error in the log, even if we plan on retrying the request. 

We should only print a one line warning about this, and print an error only if we no longer plan on retrying. We should distinguish known errors here and also remove stack trace from things like timeouts, as the stack trace has no added value. For now let's do timeout use case, next step should be to do known http status code errors, e.g. 403, 404.

This should be new default behaviour, but we want to allow printing the full errors with stack traces via some configuration. Let's make this work based on some env var (let's call it just `VERBOSE_RETRY_LOGS`, we can improve on that later if needed).

**Comment by B4nan** [comment]
More errors where we dont care about stack trace:

```
2022-02-14T12:48:24.605Z   RequestError: The proxy server rejected the request with status code 502
2022-02-14T12:48:24.611Z       at ClientRequest.<anonymous> (/usr/src/app/node_modules/got-cjs/dist/source/core/index.js:820:126)
2022-02-14T12:48:24.618Z       at Object.onceWrapper (node:events:510:26)
2022-02-14T12:48:24.623Z       at ClientRequest.emit (node:events:402:35)
2022-02-14T12:48:24.631Z       at ClientRequest.origin.emit (/usr/src/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
2022-02-14T12:48:24.644Z       at ClientRequest.request.emit (/usr/src/app/node_modules/got-scraping/dist/hooks/fix-decompress.js:76:20)
2022-02-14T12:48:24.652Z       at emitErrorNT (node:internal/streams/destroy:157:8)
2022-02-14T12:48:24.656Z       at emitErrorCloseNT (node:internal/streams/destroy:122:3)
2022-02-14T12:48:24.661Z       at processTicksAndRejections (node:internal/process/task_queues:83:21)
2022-02-14T12:48:24.665Z       at Http2OverHttp.createConnection (/usr/src/app/node_modules/http2-wrapper/source/proxies/h2-over-hx.js:19:10)
2022-02-14T12:48:24.669Z       at runMicrotasks (<anonymous>)
2022-02-14T12:48:24.673Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-02-14T12:48:24.678Z       at async entry (/usr/src/app/node_modules/http2-wrapper/source/agent.js:399:15)
```

```
2022-02-14T12:47:19.660Z   Error: Got status code 404
2022-02-14T12:47:19.680Z       at BasicCrawler.handleRequestFunction [as userProvidedHandler] (/usr/src/app/main.js:56:23)
2022-02-14T12:47:19.685Z       at runMicrotasks (<anonymous>)
2022-02-14T12:47:19.689Z       at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-02-14T12:47:19.693Z       at async BasicCrawler._handleRequestFunction (/usr/src/app/node_modules/apify/build/crawlers/basic_crawler.js:324:9)
2022-02-14T12:47:19.697Z       at async wrap (/usr/src/app/node_modules/@apify/timeout/index.js:73:27)
```

**Comment by B4nan** [comment]
Closed via #44, let's have a separate issue for follow ups.
